test/test/event.cpp

```diff
@@ -0,0 +1,98 @@
+#include "pch.h"
+
+using namespace winrt;
+using namespace Windows::Foundation;
+
+namespace
+{
+    struct Sender : implements<Sender, IStringable>
+    {
+        hstring m_value{ L"Sender" };
+
+        hstring ToString()
+        {
+            return m_value;
+        }
+    };
+}
+
+TEST_CASE("event")
+{
+    handle start{ CreateEventW(nullptr, true, false, nullptr) };
+    handle completed{ CreateEventW(nullptr, true, false, nullptr) };
+
+    auto sender = make<Sender>();
+    async_event<TypedEventHandler<IStringable, int>> event;
+
+    // No handlers and the resulting IAsyncAction completes synchronously (without blocking).
+    REQUIRE(event(sender, 123).Status() == AsyncStatus::Completed);
+
+    auto token = event.add([&](auto&& sender, auto&& args)
+        {
+            // Lengthy timeout allowing test thread to signal start.
```

Comment:
Why bother waiting at all?

Reply:
This ensures that the test can observe the the status of the async transitions from the Started to Completed states.